### PR TITLE
Add shellcheck pre-commit hook

### PR DESCRIPTION
**Pull request by jvperrin**
A bit messy on the first commit because there was a lot to fix, but `shellcheck` should keep our shell scripts more consistent and less error-prone. I'd like to submit a PR upstream to [pre-commit-shell](https://github.com/detailyang/pre-commit-shell) to support using types, because currently it only supports checking files that end in the correct extension:

    files: (\.sh|\.zsh|\.ksh)$

### CODE REVIEW

**Comment by matthew-mcallister** [comment]
Wow, this is like 10+ years of code getting linted here.

File: acct/check [hunk 1]
```diff
@@ -145,7 +145,7 @@ echo
 paper view "$user"
 
 # check User_Info entries
-User_Info=$([ -f "$User_Info_file" ] && grep ^$user: "$User_Info_file" | cut -d: -f2-)
+User_Info=$([ -f "$User_Info_file" ] && grep ^"$user": "$User_Info_file" | cut -d: -f2-)
```
Comment (chriskuehl) [inline acct/check]: this works but is kinda weird. i'd write `"^$user"` instead of `^"$user"`

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -31,3 +31,15 @@
...
+# (https://github.com/detailyang/pre-commit-shell)
+# once it supports types and not just files ending in shell extensions
+-   repo: local
+    hooks:
+    -   id: shell-lint
+        name: Shell Syntax Check
+        description: Check Shell Syntax on ALL staged files with user friendly messages and colors
+        entry: shell-lint.sh
+        language: script
+        exclude: ^.deactivate.sh$
```
Comment (chriskuehl) [inline .pre-commit-config.yaml]: shouldn't need this anymore if using types

Comment (chriskuehl) [inline .pre-commit-config.yaml]: oh, i skimmed too fast. yeah, you probably need this

Comment (chriskuehl) [inline .pre-commit-config.yaml]: (i assumed you were excluding the .activate.sh symlink rather than the .deactivate.sh file)

**Comment by jvperrin** [comment]
We do [install `shellcheck` in extrapackages](https://github.com/ocf/puppet/blob/aef2a578f316c9eb09b624746b29eeb7f33543ed/modules/ocf/manifests/extrapackages.pp#L161), but yeah, it won't work everywhere, so that's a good point.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (jvperrin) [inline .pre-commit-config.yaml]: Ah, good point, thanks. I put this there before using types, so that's better

File: acct/check (hunk 1 above)
Comment (jvperrin) [inline acct/check]: Haha yeah, that would be better, too many changes to make that I didn't look at this very closely

File: .pre-commit-config.yaml (hunk 2 above)
Comment (jvperrin) [inline .pre-commit-config.yaml]: Still seems to work without it, so that's fine, I'll remove it.

**Pull request merged**